msprime/demography.py

```diff
@@ -58,6 +59,75 @@ class IncompletePopulationMetadataWarning(UserWarning):
     """
 
 
+class LruCache(collections.OrderedDict):
+    # LRU example from the OrderedDict documentation
+    def __init__(self, maxsize=128, *args, **kwds):
+        self.maxsize = maxsize
+        super().__init__(*args, **kwds)
+
+    def __getitem__(self, key):
+        value = super().__getitem__(key)
+        self.move_to_end(key)
+        return value
+
+    def __setitem__(self, key, value):
+        super().__setitem__(key, value)
+        if len(self) > self.maxsize:
+            oldest = next(iter(self))
+            del self[oldest]
+
+
+_population_table_cache = LruCache(16)
+
+
+def _build_population_table(populations):
+    """
+    Return a tskit PopulationTable instance encoding the metadata for the
+    specified populations. Because encoding metadata is quite expensive
+    we maintain an LRU cache.
+    """
+    key = json.dumps([pop.asdict() for pop in populations], sort_keys=True)
```

Comment:
Hang on, doesn't Population.asdict() include demographic information, not just name/description/metadata as used here?

Reply:
Yes, you're right - I guess if you want to make the minimal key you'd just build the actual population metadata items and cache on them.

Sigh - I'll update.

Reply:
OK, updated - how does it look now?

Reply:
Yes, that looks right now. Thanks!